### ISSUE TITLE
7$: Improve error for `else (cond) {}`

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#578]
https://stackoverflow.com/questions/41942368/uncaught-syntaxerror-unexpected-token
https://stackoverflow.com/questions/38105591/syntaxerror-unexpected-token

I think this should be easy to do. In `parse_and_visit_if`, after parsing the body of the `else`, we can check if the next token is `{` with `has_leading_newline == false`. If so, report an error.

**Comment by singalhimanshu** [comment]
> if the next token is `{`

shouldn't we report an error if the next token is '(' ?

**Comment by strager** [comment]
> > if the next token is `{`
> 
> shouldn't we report an error if the next token is `(`?

Both, actually.

If the token after `else` is `(` *and* if the token after the body is `{` on the same line, then report an error.

**Comment by singalhimanshu** [comment]
> _and_ if the token after the body is `{` on the same line

I didn't understand this one. Can you give an example?

**Comment by strager** [comment]
No error:
```js
if (x) banana();
else (orange())
{
  stuff();
}
```

No error:
```js
if (x) banana();
else (orange())
stuff();
```

Error:
```js
if (x) banana();
else (orange()) {
  stuff();
}
```

**Comment by singalhimanshu** [comment]
> ```js
> if (x) banana();
> else (orange()) {
>   stuff();
> }
> ```

What should the error highlight in this case?

**Comment by strager** [comment]
> What should the error highlight in this case?

Between `else` and `(`. "missing 'if' after 'else'"

**Comment by strager** [comment]
@singalhimanshu Your patch had a few mistakes which I had to fix myself. See commits d9339b90fd85bf996ed0da81530c6326cd1e63e6 and ce2ba53fade53bf00392258995fce12f7c9113f5.
